src/pgduckdb_duckdb.cpp

```diff
@@ -156,6 +178,18 @@ DuckDBManager::Initialize() {
 
 	LoadFunctions(context);
 	LoadExtensions(context);
+
+	RegisterDuckdbXactCallback();
```

Comment:
TODO: This should be moved to PG_init (or backend start). Now that we use this to reset the `top_level_statement` variable it should always be enabled and never unregistered.